Crash caused when json file is access via http (instead of https)

If the JSON version file is stored on an http server and is not available via https, then the library will cause a crash like this:
```
01-24 11:36:58.578 28106-28297/com.example E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
     Process: com.example, PID: 28106
     java.lang.RuntimeException: An error occurred while executing doInBackground()
         at android.os.AsyncTask$3.done(AsyncTask.java:325)
         at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
         at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
         at java.util.concurrent.FutureTask.run(FutureTask.java:242)
         at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
         at java.lang.Thread.run(Thread.java:761)
      Caused by: java.lang.ClassCastException: com.android.okhttp.internal.huc.HttpURLConnectionImpl cannot be cast to javax.net.ssl.HttpsURLConnection
         at com.eggheadgames.siren.Siren$LoadJsonTask.doInBackground(Siren.java:266)
         at com.eggheadgames.siren.Siren$LoadJsonTask.doInBackground(Siren.java:258)
         at android.os.AsyncTask$2.call(AsyncTask.java:305)
         at java.util.concurrent.FutureTask.run(FutureTask.java:237)
         at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243) 
         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
         at java.lang.Thread.run(Thread.java:761) 

```

This seems to have been introduced by #10 / #11 whereby the code now only works if the JSON file is fetched via https.

We should make two fixes:

- [ ] allow an `http` url to be fetched (not just `https`)
- [ ] silently fail, except for a debug log message, instead of crashing, if there is an error fetching the json file

--- a/library/build.gradle
+++ b/library/build.gradle
@@ -11,8 +11,8 @@ android {
     defaultConfig {
         minSdkVersion 15
         targetSdkVersion 25
-        versionCode 7
-        versionName "1.4.6"
+        versionCode 8
+        versionName "1.4.7"
     }
     buildTypes {
         release {

--- a/library/src/main/java/com/eggheadgames/siren/Siren.java
+++ b/library/src/main/java/com/eggheadgames/siren/Siren.java
@@ -12,9 +12,8 @@ import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.lang.ref.WeakReference;
+import java.net.HttpURLConnection;
 import java.net.URL;
-import java.security.KeyManagementException;
-import java.security.NoSuchAlgorithmException;
 
 import javax.net.ssl.HttpsURLConnection;
 
@@ -259,17 +258,19 @@ public class Siren {
 
         @Override
         protected String doInBackground(String... params) {
-            HttpsURLConnection connection = null;
+            HttpURLConnection connection = null;
             try {
                 TLSSocketFactory TLSSocketFactory = new TLSSocketFactory();
                 URL url = new URL(params[0]);
-                connection = (HttpsURLConnection) url.openConnection();
+                connection = (HttpURLConnection) url.openConnection();
                 connection.setRequestMethod("GET");
                 connection.setUseCaches(false);
                 connection.setAllowUserInteraction(false);
                 connection.setConnectTimeout(10000);
                 connection.setReadTimeout(10000);
-                connection.setSSLSocketFactory(TLSSocketFactory);
+                if ("https".equalsIgnoreCase(url.getProtocol())) {
+                    ((HttpsURLConnection)connection).setSSLSocketFactory(TLSSocketFactory);
+                }
                 connection.connect();
                 int status = connection.getResponseCode();
 
@@ -298,8 +299,8 @@ public class Siren {
                     Siren.sirenInstance.mSirenListener.onError(ex);
                 }
 
-            } catch (NoSuchAlgorithmException | KeyManagementException e) {
-              e.printStackTrace();
+            } catch (Exception ex) {
+              ex.printStackTrace();
             } finally {
                 if (connection != null) {
                     try {

